### PR TITLE
Port more tools to vertical table

### PR DESCRIPTION
**Pull request by dtardon**
Note: I'm not sure if the first commit is worth it for just 2 uses, but I've left it in for now...

### CODE REVIEW

File: src/shared/format-table.h [hunk 1]
```diff
@@ -88,7 +88,29 @@ int table_add_cell_full(Table *t, TableCell **ret_cell, TableDataType type, cons
 static inline int table_add_cell(Table *t, TableCell **ret_cell, TableDataType type, const void *data) {
         return table_add_cell_full(t, ret_cell, type, data, SIZE_MAX, SIZE_MAX, UINT_MAX, UINT_MAX, UINT_MAX);
 }
-int table_add_cell_stringf(Table *t, TableCell **ret_cell, const char *format, ...) _printf_(3, 4);
+int table_add_cellv(Table *t, TableCell **ret_cell, TableDataType type, const char *format, va_list ap) _printf_(4, 0);
+_printf_(4, 5) static inline int table_add_cellf(Table *t, TableCell **ret_cell, TableDataType type, const char *format, ...)
+{
```
Comment (yuwata) [inline src/shared/format-table.h]: Coding style: please move `{` at the end of the previous line.

File: src/shared/format-table.h [hunk 2]
```diff
@@ -88,7 +88,29 @@ int table_add_cell_full(Table *t, TableCell **ret_cell, TableDataType type, cons
...
+        int r;
+
+        va_start(ap, format);
+        r = table_add_cellv(t, ret_cell, type, format, ap);
+        va_end(ap);
+
+        return r;
+}
+_printf_(3, 4) static inline int table_add_cell_stringf(Table *t, TableCell **ret_cell, const char *format, ...)
+{
```
Comment (yuwata) [inline src/shared/format-table.h]: Also here.

**Review by yuwata — APPROVED** [review]
LGTM, except for the two minor coding style issues.

File: src/shared/format-table.h (hunk 1 above)
Comment (dtardon) [reply src/shared/format-table.h]: I removed the inlining completely, by changing `table_add_cell_stringf` into a macro that expands to `table_add_cellf`. Which is really what I wanted to do from the beginning, just didn't think about it...

File: src/shared/format-table.c [hunk 3]
```diff
@@ -512,18 +512,20 @@ int table_add_cell_full(
...
 }
 
-int table_add_cell_stringf(Table *t, TableCell **ret_cell, const char *format, ...) {
+int table_add_cellf(Table *t, TableCell **ret_cell, TableDataType type, const char *format, ...) {
         _cleanup_free_ char *buffer = NULL;
         va_list ap;
         int r;
 
         va_start(ap, format);
+
```
Comment (yuwata) [inline src/shared/format-table.c]: (Please do not insert unnecessary empty line.)

File: src/shared/format-table.c [hunk 4]
```diff
@@ -512,18 +512,20 @@ int table_add_cell_full(
...
         r = vasprintf(&buffer, format, ap);
-        va_end(ap);
         if (r < 0)
                 return -ENOMEM;
 
-        return table_add_cell(t, ret_cell, TABLE_STRING, buffer);
+        va_end(ap);
```
Comment (yuwata) [inline src/shared/format-table.c]: This must be called before exiting from this function even on error:
```
va_start();
r = vasprintf();
va_end();
if (r < 0)
        return -ENOMEM;
```

**Review by yuwata — CHANGES_REQUESTED** [review]

**Review by yuwata — APPROVED** [review]
Rebased.

**Pull request merged**